pkg/odo/cli/delete/component.go

```diff
@@ -91,11 +99,85 @@ func (o *ComponentOptions) deleteNamedComponent() error {
 	return nil
 }
 
-// deleteDevfileComponent deletes a component defined by the devfile in the current directory
+// printDevfileResources prints the devfile components for deleteDevfileComponent
+func (o *ComponentOptions) printDevfileComponents(devfileObj parser.DevfileObj) (undeploy bool, err error) {
+	log.Info("The following resources will be deleted: ")
+	log.Printf("Devfile component: %v", devfileObj.GetMetadataName())
+
+	resources, err := libdevfile.ListKubernetesComponents(devfileObj, filepath.Dir(o.EnvSpecificInfo.GetDevfilePath()))
+	if err != nil {
+		return false, fmt.Errorf("Failed to gather resources for deletion: %w", err)
+	}
+	if len(resources) != 0 {
+		log.Printf("Kubernetes resources:")
+		for _, resource := range resources {
+			fmt.Println("\t", resource.GetKind(), ":", resource.GetName())
+		}
+		undeploy = true
+	}
+	return
+}
+
+// deleteDevfileComponent deletes all the components defined by the devfile in the current directory
 func (o *ComponentOptions) deleteDevfileComponent() error {
+	devfileObj := o.EnvSpecificInfo.GetDevfileObj()
+
+	// Print all the resources that will be deleted.
+	performUndeploy, err := o.printDevfileComponents(devfileObj)
```

Comment:
Having this function `printDevfileComponents` to also compute `undeploy` is confusing, as these 2 things are non related from the caller point of view.

I would instead have a separate function to check if there are Kubernetes components in a devfile, that would be more eloquent

Reply:
Yeah, that makes more sense. On it.